### PR TITLE
Do not lose focus after remove table

### PR DESCRIPTION
**Pull request by zhujinxuan**
This is a PR about focus selection to the next or prev block when the table is removed;

There is a bit trick about undo, if change follows the sequence that `remove_node`=>`set_selection`, and then focus will be restored by undo.  However, the selection range is still not well restored.

### CODE REVIEW

File: lib/changes/removeTable.js [hunk 1]
```diff
@@ -13,8 +13,40 @@ function removeTable(opts: Options, change: Change): Change {
...
-    return change.removeNodeByKey(table.key);
+    const nextBlock = change.value.document.getNextBlock(table.key);
+    if (nextBlock) {
+        nextFocusBlock = nextBlock;
+    } else {
+        const prevBlock = change.value.document.getPreviousBlock(table.key);
+        if (prevBlock) {
+            nextFocusBlock = prevBlock;
+            shouldCollapseToEnd = true;
+        } else {
```
Comment (CameronAckermanSEL) [inline lib/changes/removeTable.js]: potentially should just not handle the condition where there's no other block to select. Some users will pass null for the exitBlockType to disable that feature.

**Comment by Soreine** [comment]
Looks great 👍 
Would you be able to add a corresponding test ?

Concerning undo, I would have guessed that...

``` 
A => set_selection => B => remove_node => C
```

... would allow a proper undo, because B would have a valid selection.

The following...

```
A => remove_node  => B => set_selection => C
```

... has B in an invalid state. So undoing from C to B would cause an error ?

**Comment by zhujinxuan** [comment]
@Soreine 
I will make the test at this weekends.  
Undo remove table cannot restore selection.  It would pass selection to the end of node previous to table.  I am thinking about add a fake selection operation to restore the selection

**Comment by zhujinxuan** [comment]
@Soreine  Almost done with the test.  But there is one function I am not sure how to test: 

The feature is, when the document contains only one table and nothing else and the `exitBlock===null` in plugin, the document will be empty without any focus.

**Comment by Soreine** [comment]
Great ! For that edge case, this plugin cannot do much. It would be worth making a plugin that handles empty documents cases like this and normalize them instead.

**Comment by zhujinxuan** [comment]
@Soreine The `exitBlock === null` test is added by the recent commit d6667bb in the PR.  I think the tests are done.

Would you like to publish a new version after merging this PR?
Thanks.

**Pull request merged**